src/services/infra/InfraService.ts

```diff
@@ -324,32 +340,104 @@ export default class InfraService {
       }
 
       this.queueService.sendMessage(message)
+
+      return ok(newLaunchParams)
     } catch (error) {
-      logger.error(`Failed to created '${repoName}' site on Isomer: ${error}`)
-      return err(error)
+      logger.error(`Failed to create '${repoName}' site on Isomer: ${error}`)
+      // requester email is guaranteed to exist as currently these are Isomer users
+      this.sendRetryToIsomerAdmin(requestor.email!, repoName)
+
+      throw error
     }
-    return ok(newLaunchParams)
+  }
+
+  sendRetryToIsomerAdmin = async (email: string, repoName: string) => {
+    const subject = `[Isomer] Failure to create domain association for ${repoName}`
+    const body = `<p>Unable to trigger create domain association for ${repoName}.</P
+    <p>If domain association was already created, please log into the amplify console and trigger a retry. </p>
+    <p>Else, resubmit the form and try again.</p>`
+    await mailer.sendMail(email, subject, body)
   }
 
   siteUpdate = async () => {
-    try {
-      const messages = await this.queueService.pollMessages()
-      await Promise.all(
-        messages.map(async (message) => {
-          const site = await this.sitesService.getBySiteName(message.repoName)
-          if (site) {
-            const updateSuccessSiteLaunchParams = {
+    const messages = await this.queueService.pollMessages()
+    await Promise.all(
+      messages.map(async (message) => {
+        const site = await this.sitesService.getBySiteName(message.repoName)
+        if (site) {
```

Comment:
seems to have no `else`? maybe we should just do an early return and avoid an extra level of indentation.

Reply:
opps, yup, changed 8138374